### PR TITLE
feat: add gochck architecture linter

### PR DESCRIPTION
**Pull request by alexandreliberato**
Added a config file for gochk tool, changed Makefile to get the tool and added a new step to execute it.

Type: feat

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,7 +25,8 @@ setup:
 	github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway \
     github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2 \
     golang.org/x/tools/cmd/goimports \
-    github.com/kyoh86/richgo
+    github.com/kyoh86/richgo \
+    github.com/resotto/gochk
```
Comment (alexrios) [inline Makefile:29]: Please, add `gochk` on `tools.go` to guarantee the same version to everybody using the project.

File: gochk-arch-linter.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencyOrders": ["external", "adapter", "application", "domain"],
```
Comment (alexrios) [inline gochk-arch-linter.json:2]: `dependencyOrders` doesn't match the current folders, right?

File: Makefile [hunk 3]
```diff
@@ -61,6 +62,11 @@ metalint:
 	golangci-lint run -c ./.golangci.yml ./...
 	buf check lint
 
+.PHONY: archlint
```
Comment (alexrios) [inline Makefile:65]: This step is a good candidate to be appended on `pre/push` step.

File: gochk-arch-linter.json (hunk 2 above)
Comment (alexandreliberato) [reply gochk-arch-linter.json]: sure! it's corrected now.

File: Makefile (hunk 3 above)
Comment (alexgarzao) [reply Makefile:65]: Yep. It's a good idea.

**Review by alexgarzao — APPROVED** [review]

**Comment by alexandreliberato** [comment]
@alexgarzao I think that we just need to merge now :)

**Comment by alexgarzao** [comment]
> @alexgarzao I think that we just need to merge now :)

Yep. Very nice contribution :-)

**Pull request merged**